### PR TITLE
swift-module-digester: diagnose adding variables with fixed binary order.

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

File: include/swift/AST/DiagnosticsModuleDiffer.def [hunk 1]
```diff
@@ -54,6 +54,8 @@ ERROR(decl_new_attr,none,"%0 is now %1", (StringRef, StringRef))
 
 ERROR(decl_reorder,none,"%0 in a fixed layout type changes position from %1 to %2", (StringRef, unsigned, unsigned))
 
+ERROR(decl_added,none,"%0 is added to a non-resilient type", (StringRef))
```
Comment (slavapestov) [inline include/swift/AST/DiagnosticsModuleDiffer.def]: Please change the wording to be consistent (either always call them non-resilient, or fixed layout/frozen)

**Review by slavapestov — COMMENTED** [review]
Can you test lazy properties too?

File: include/swift/AST/DiagnosticsModuleDiffer.def (hunk 1 above)
Comment (nkcsgexi) [reply include/swift/AST/DiagnosticsModuleDiffer.def]: I've seen new diags like "cake2: Var fixedLayoutStruct.lazyD.storage is added to a non-resilient type" for lazy var. Is this expected?

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Pull request merged**